### PR TITLE
# Use Boost library in SDL

### PR DESCRIPTION
**Pull request by AGaliuzov**
The proposal is to use Boost library in C++.

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
Please add specify proposed boost version

File: proposals/NNNN-use-Boost-library.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## Motivation
+From time to time we’re facing with different problems with our [Utils](https://github.com/smartdevicelink/sdl_core/tree/master/src/components/utils/include/utils) module - threads, timers, file system.
+* We've faced with different issues during porting to Windows platform because items mentioned above were designed in platform specific manner. We had to refactor existing architectural solutions, invent workarounds to fix platform/compiler specific issues.
+* All the time we have to solve customer’s problems very fast and we have no enough time for current utilities adaptation. Usually it much faster to develop necessary functionality from the scratch. But such approach brings us new defects, waste our time to support and improve this functionality. 
+
+## Proposed solution
+The porposed solution whic has to adress mentioned issues is to use [Boost library](http://www.boost.org)
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-use-Boost-library.md]: whic**h**

File: proposals/NNNN-use-Boost-library.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+* The way to solve the issue is to use Boost library. It supports dozens of platforms out of the box, even with ancient compilers.
+* The Boost license encourages both commercial and non-commercial use. The Boost license permits the creation of derivative works for commercial or 	non-commercial 	use with no legal requirement to release your source 	code. Other differences include Boost not requiring reproduction of 	copyright messages for object code redistribution, and the fact that the Boost license is not "viral": if you distribute your own code along with some 	Boost code, the Boost license applies only to the Boost code (and 	modified versions thereof); you are free to 	license your own code under any terms you like. The GPL is also much longer, and thus may be 	harder to 	understand.
+##### Benifits
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-use-Boost-library.md]: @AGaliuzov reduce amount of `#`

**Review by LuxoftAKutsan — APPROVED** [review]

File: proposals/NNNN-use-Boost-library.md (hunk 2 above)
Comment (AGaliuzov) [reply proposals/NNNN-use-Boost-library.md]: @LuxoftAKutsan why?

**Pull request closed without merging**